src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -121,23 +168,41 @@ const useFetchChannelHook = () => {
   const saveAllChannelData = async (channels, channelCategory: ChannelCategory) => {
     const filteredChannels = filterChannels(channels);
 
-    filteredChannels?.forEach(async (channel) => {
-      await saveChannelData(channel, channelCategory);
+    const channelPromises = filteredChannels?.map((channel) => {
+      return new Promise((resolve, reject) => {
+        try {
+          saveChannelData(channel, channelCategory).then(() => resolve(true));
+        } catch (error) {
+          reject(error);
+        }
+      });
+    });
+
+    queue.addJob({
+      label: 'timestamp update',
+      task: async () => {
+        const timestamp = new Date().toISOString();
+        StorageUtils.channelSignedTimeStamps.set(timestamp);
+      }
     });
+
+    await Promise.all(channelPromises);
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here introduce a queue system for saving all channel data and updating timestamps. This is a good approach for improving performance by allowing these operations to be processed asynchronously. However, similar to the previous comments, consider adding error handling within each job task to reject the promise with the error when an issue occurs.



<!-- This is an auto-generated comment by OSS CodeRabbit -->